src/server/string_family.cc

```diff
@@ -525,13 +525,32 @@ SinkReplyBuilder::MGetResponse OpMGet(bool fetch_mcflag, bool fetch_mcver, const
   response.storage_list = SinkReplyBuilder::AllocMGetStorage(total_size);
   char* next = response.storage_list->data;
 
+  auto log_error = [](auto it) {
+    LOG(ERROR) << "Invalid type " << it->second.ObjType();
+    LOG(ERROR) << "Bucket id: " << it.bucket_id();
+    LOG(ERROR) << "Segment " << it.segment_id();
+    LOG(ERROR) << "Slot id " << it.slot_id();
+  };
+
   for (size_t i = 0; i < args.size(); ++i) {
     PrimeConstIterator it = iters[i];
     if (it.is_done())
       continue;
 
     auto& resp = response.resp_arr[i].emplace();
-
+    if (it->second.ObjType() != OBJ_STRING) {
```

Comment:
1. I would like to print all the valid iterators in  iters, not just the one that is inconsistent
2. I actually want to test two conditions: objtype like you do here and also it->first vs `args[i]`. i.e. if any of them do not hold, you trigger the fallback.
3. The fallback should also call again `db_slice.FindReadOnly` and find the correct iterator and continue with the normal flow of checking again  for ObjType and it.done and proceeding with CopyValueToBuffer again.

Reply:
The code should not be beautiful, but should be correct. It's a temporary code.